src/content_mapblock.cpp

```diff
@@ -1288,14 +1288,20 @@ void MapblockMeshGenerator::errorUnknownDrawtype()
 
 void MapblockMeshGenerator::drawNode()
 {
+	// skip some drawtypes early
+	switch (f->drawtype) {
+		case NDT_NORMAL:   // Drawn by MapBlockMesh
+		case NDT_AIRLIKE:  // Not drawn at all
+		case NDT_LIQUID:   // Drawn by MapBlockMesh
+			return;
+		default:
+	}
+	origin = intToFloat(p, BS);
```

Comment:
Actually, this referred to the member `p` before as well... So on second thought it looks correct.